src/python/pants/backend/python/goals/package_dists.py

```diff
@@ -9,19 +9,27 @@
 from pants.backend.python.util_rules.package_dists import create_dist_build_request
 from pants.backend.python.util_rules.package_dists import rules as package_dists_rules
 from pants.core.goals.package import BuiltPackage, BuiltPackageArtifact, PackageFieldSet
+from pants.engine.internals.native_engine import Address
 from pants.engine.intrinsics import digest_to_snapshot
 from pants.engine.rules import collect_rules, implicitly, rule
 from pants.engine.unions import UnionMembership, UnionRule
 
 
 @rule
-async def package_python_dist(
+async def dist_target_address(
     field_set: PythonDistributionFieldSet,
+) -> Address:
+    return field_set.address
+
+
+@rule
+async def package_python_dist(
+    dist_target_address: Address,
```

Comment:
I will note that this means we can't have a rule in some other backend that goes from Address to BuiltPackage, if accessing via Get (via call-by-name is fine of course). So this is only intended to be used via CBN.